compiler/rustc_middle/src/mir/visit.rs

```diff
@@ -998,12 +998,11 @@ macro_rules! visit_place_fns {
     () => {
         fn visit_projection(
             &mut self,
-            local: Local,
-            projection: &[PlaceElem<'tcx>],
+            place_ref: PlaceRef<'tcx>,
             context: PlaceContext,
             location: Location,
         ) {
-            self.super_projection(local, projection, context, location);
+            self.super_projection(place_ref.local, place_ref.projection, context, location);
```

Comment:
This looks like `super_projection` should get the same treatment, can you also do that? The `super_` and `visit_` methods should be consistent with each other.